src/js/controls/Header.js

```diff
@@ -266,31 +545,150 @@ lm.utils.copy( lm.controls.Header.prototype, {
 	 *
 	 * @returns {void}
 	 */
-  	_showAdditionalTabsDropdown: function(e) {
+  	_showAdditionalTabsDropdown: function() {
+		if (this.isDropdownShown) {
+			this._hideAdditionalTabsDropdown();
+			return;
+		}
+
+		// clone tabs in the current list, with event listeners
+		// and add them to the drop down
+		this.tabDropdownList = 
+			this.tabsContainer.clone(true)
+				.appendTo( this.tabDropdownContainer )
+				.children().removeClass('lm_active');
+
+		// show the dropdown
 		this.tabDropdownContainer.show();
-		this.tabDropdownButton.element.off(); // take off the click toggle
-		$(document).on("mouseup", this.hideAdditionalTabsDropdown);
-		this._updateAdditionalTabsDropdown();
+		this.isDropdownShown = true;
+
+		// focus the dropdown filter list input
+		this.tabDropdownSearch.val('').focus();
+		this.dropdownKeyIndex = 0;
+		this.tabDropdownList.eq(this.dropdownKeyIndex).addClass('lm_keyboard_active');
+		
+		$(document).on('mousedown', this._hideAdditionalTabsDropdown);
+		this._updateAdditionalTabsDropdown();		
+	},
+
+	// enables synthetic keyboard navigation of the list
+	_handleFilterKeydown: function(e) {
+		var i;
+
+		if(this.dropdownKeyIndex === -1) return;
+
+		if(e.key === 'Escape') {
+			this._hideAdditionalTabsDropdown();
+			return;
+		}
+		
+		if (e.key === 'Enter') {
+			// simulate "click"
+			this._hideAdditionalTabsDropdown();
+			this.tabs[this.dropdownKeyIndex]._onTabClick();
+			return;
+		}
+		
+		var newIndex = -1;
+		if (e.key === 'ArrowDown') {
+			this.tabDropdownList.eq(this.dropdownKeyIndex).removeClass('lm_keyboard_active');
+			// find the next visible index
+			for (i = this.dropdownKeyIndex + 1 ; i < this.tabDropdownList.length; i++) {
```

Comment:
Could optimize this bit...

```
function getNextDropdownIndex(startIndex, delta) {
  if (this.tabDropdownList.length < 2) {
    return -1;
  }
  let i = (startIndex + delta + this.tabDropdownList.length) % this.tabDropdownList.length;
  while (i != startIndex) {
    if ( this.tabDropdownList.eq(i).css('display') !== 'none' ) {
      return i;
    }
    i = (i + delta + this.tabDropdownList.length) % this.tabDropdownList.length;
  }

  return startIndex;
}
```
Then just:
```
if (e.key === 'ArrowDown') {
  this.dropdownKeyIndex = getNextDropdownIndex(this.dropdownKeyIndex, 1);
} else if (e.key === 'ArrowUp') {
  this.dropdownKeyIndex = getNextDropdownIndex(this.dropdownKeyIndex, -1);
}
```

Reply:
that's much more clever.